python/lsst/faro/utils/stellarLocus.py

```diff
@@ -0,0 +1,232 @@
+        bPerpMax : `float`
+            The intercept of the perpendicular line that goes through xMax
+        mODR2 : `float`
+            The gradient from the second round of fitting
+        bODR2 : `float`
+            The intercept from the second round of fitting
+        mPerp : `float`
+            The gradient of the line perpendicular to the line from the
+            second fit
+    Notes
+    -----
+    The code does two rounds of fitting, the first is initiated using the
+    hardwired values given in the `paramDict` parameter and is done using
+    an Orthogonal Distance Regression fit to the points defined by the
+    box of xMin, xMax, yMin and yMax. Once this fitting has been done a
+    perpendicular bisector is calculated at either end of the line and
+    only points that fall within these lines are used to recalculate the fit.
+    """
+
+    # Points to use for the fit
+    fitPoints = np.where((xs > paramDict["xMin"]) & (xs < paramDict["xMax"])
+                         & (ys > paramDict["yMin"]) & (ys < paramDict["yMax"]))[0]
+
+    linear = scipyODR.polynomial(1)
+
+    data = scipyODR.Data(xs[fitPoints], ys[fitPoints])
+    odr = scipyODR.ODR(data, linear, beta0=[paramDict["bHW"], paramDict["mHW"]])
+    params = odr.run()
+    mODR = float(params.beta[1])
+    bODR = float(params.beta[0])
+
+    paramsOut = {"xMin": paramDict["xMin"], "xMax": paramDict["xMax"], "yMin": paramDict["yMin"],
+                 "yMax": paramDict["yMax"], "mHW": paramDict["mHW"], "bHW": paramDict["bHW"],
+                 "mODR": mODR, "bODR": bODR}
+
+    # Having found the initial fit calculate perpendicular ends
+    mPerp = -1.0/mODR
+    # When the gradient is really steep we need to use
+    # the y limits of the box rather than the x ones
+
+    if np.fabs(mODR) > 1:
+        yBoxMin = paramDict["yMin"]
+        xBoxMin = (yBoxMin - bODR)/mODR
+        yBoxMax = paramDict["yMax"]
+        xBoxMax = (yBoxMax - bODR)/mODR
+    else:
+        yBoxMin = mODR*paramDict["xMin"] + bODR
+        xBoxMin = paramDict["xMin"]
+        yBoxMax = mODR*paramDict["xMax"] + bODR
+        xBoxMax = paramDict["xMax"]
+
+    bPerpMin = yBoxMin - mPerp*xBoxMin
+
+    paramsOut["yBoxMin"] = yBoxMin
+    paramsOut["bPerpMin"] = bPerpMin
+
+    bPerpMax = yBoxMax - mPerp*xBoxMax
+
+    paramsOut["yBoxMax"] = yBoxMax
+    paramsOut["bPerpMax"] = bPerpMax
+
+    # Use these perpendicular lines to chose the data and refit
+    fitPoints = ((ys > mPerp*xs + bPerpMin) & (ys < mPerp*xs + bPerpMax))
+    data = scipyODR.Data(xs[fitPoints], ys[fitPoints])
+    odr = scipyODR.ODR(data, linear, beta0=[bODR, mODR])
+    params = odr.run()
+    mODR = float(params.beta[1])
+    bODR = float(params.beta[0])
+
+    paramsOut["mODR2"] = float(params.beta[1])
+    paramsOut["bODR2"] = float(params.beta[0])
+
+    paramsOut["mPerp"] = -1.0/paramsOut["mODR2"]
+
+    # Calculate the fit lines for the entire X, Y range
```

Comment:
I'm not sure I understand why this is necessary.

Reply:
And it doesn't redo the fitting so the comment is a little misleading, it just moves the points on the line that you use for the distance calculations? I am very confused by this.

Reply:
It is applying the coefficients to the min/max x and y values so that these points can be used to calculate the distance from the fit line. (i.e., it is "calculating" the values of line endpoints) I'll see if I can clarify the docstring.